lib/std/macho.zig

```diff
@@ -890,6 +890,10 @@ pub const nlist_64 = extern struct {
         if (!sym.undf()) return false;
         return sym.n_value != 0;
     }
+
+    pub fn getDylibOrdinal(sym: nlist_64) i16 {
+        return @divTrunc(@bitCast(i16, sym.n_desc), @as(u15, N_SYMBOL_RESOLVER));
+    }
```

Comment:
please avoid doing this refactoring in the same commit as the compiler changes so that the necessary changes to account for the new type system rules are visible in this commit diff. `N_SYMBOL_RESOLVER` is comptime known, so this `@as` is incorrect.